src/gnn_reco/components/loss_functions.py

```diff
@@ -73,12 +73,12 @@ def _log_cosh(cls, x: Tensor) -> Tensor:  # pylint: disable=invalid-name
         Used to avoid `inf` for even moderately large differences.
         See [https://github.com/keras-team/keras/blob/v2.6.0/keras/losses.py#L1580-L1617]
         """
-        return x + torch.nn.functional.softplus(-2. * x) - np.log(2.0)
+        return torch.log(torch.cosh(x))#x + torch.nn.functional.softplus(-2. * x) - np.log(2.0)
```

Comment:
I don't think we should replace the original: The proposed formula produce invalid result for even moderately large `x`; and for the values where the proposed formula does produce valid results, the two expressions are numerically identical, as ensured by the (now failing) unit test here: https://github.com/icecube/gnn-reco/blob/main/tests/test_loss_functions.py#L38